tlatools/org.lamport.tlatools/src/recomp/RecompVerify.java

```diff
@@ -47,151 +47,172 @@
+		String printMsg = "";
+		PerfTimer timer = new PerfTimer();
+		SymbolTable.init();
+
+		final String noInvsCfg = "no_invs.cfg";
+
+		// decompose the spec into as many components as possible
+		final List<String> components = Composition.symbolicCompose(tla, cfg, recompType, recompFile, rawComponents);
+		Utils.assertTrue(rawComponents.size() > 0, "Decomposition returned no components");
+		Utils.assertTrue(components.size() > 0, "Symbolic composition returned no components");
+		printMsg = Utils.addPrint(printMsg, "n: " + rawComponents.size());
+		printMsg = Utils.addPrint(printMsg, ("m: " + (components.size() - 1)));
+
+		// model check the first component
+		final String firstComp = components.get(0);
+		printMsg = Utils.addPrintVerbose(printMsg, verbose, "");
+		printMsg = Utils.addPrintVerbose(printMsg, verbose, "Component 1" + ": " + firstComp);
+		TLC tlcFirstComp = new TLC();
+		timer.reset();
+		tlcFirstComp.modelCheckOnlyGoodStates(firstComp, cfg); // TODO there's really no reason to distinguish between the 2 methods
+		printMsg = Utils.addPrintVerbose(printMsg, verbose, "State space gen: " + timer.timeElapsed() + "ms");
+		Utils.assertNotNull(tlcFirstComp.getLTSBuilder(), "Error generating state space for the first component!");
+
+		// turn the first component into a safety property (interface requirement)
+		timer.reset();
+		LTS<Integer, String> ltsProp = tlcFirstComp.getLTSBuilder().toIncompleteDetAutIncludingAnErrorState();
+		printMsg = Utils.addPrintVerbose(printMsg, verbose, "LTS gen: " + timer.timeElapsed() + "ms");
+		printMsg = Utils.addPrintVerbose(printMsg, verbose, "# unique states: " + (ltsProp.size()-1) + " states");
+		int totalSumOfStatesChecked = ltsProp.size() - 1;
+		int largestProductOfStatesChecked = ltsProp.size() - 1;
+		//System.out.println();
+		//FSPWriter.INSTANCE.write(System.out, ltsProp);
+		//System.out.println();
+
+		// minimize the LTS
+		timer.reset();
+		ltsProp = AutomataLibUtils.minimizeLTS(ltsProp);
+		printMsg = Utils.addPrintVerbose(printMsg, verbose, "minimization: " + timer.timeElapsed() + "ms");
+		printMsg = Utils.addPrintVerbose(printMsg, verbose, "# unique states post-minimization: " + (ltsProp.size()-1) + " states");
+
+		// initialize the alphabet
+		AlphabetMembershipTester alphabetTester = new AlphabetMembershipTester(tlcFirstComp.actionsInSpec(), ltsProp);
+
+		if (SafetyUtils.INSTANCE.ltsIsSafe(ltsProp)) {
+			final int totalNumStatesChecked = Math.max(totalSumOfStatesChecked, largestProductOfStatesChecked);
+			printMsg = Utils.addPrintVerbose(printMsg, verbose, "");
+			printMsg = Utils.addPrint(printMsg, ("k: " + 0));
+			printMsg = Utils.addPrint(printMsg, ("Total # states checked: " + totalNumStatesChecked));
+			printMsg = Utils.addPrint(printMsg, ("Property satisfied!"));
+			if (!RVStrategy.isThreadDone()) System.out.print(printMsg);
```

Comment:
When writing 'if' statements, please always use curly braces. Also, please put the "System.out.print(printMsg);" onto a new line.